flake.nix

```diff
@@ -118,6 +123,35 @@
             };
           });
         };
+      overlay_invoke = nixpkgs: pythonPackages:
+        let
+          ifNotMinVersion = pkg: ver: overlay: if (
```

Comment:
Please correct me if I'm wrong - the idea behind this overlay is to override some python packages (that are coming from nixpkgs) with newer versions, because 2.3.5 requires newer versions? If that's what you meant, I wonder, have you tried to update nixpkgs revision in a flake.lock ? Because revision of nixpkgs locked in this repo is 6 month old by now. I checked that both packages in this overlay has been updated in stable channel of 23.05 . Maybe we can just update locked nixpkgs.

Reply:
Exactly! Well, I actually tried... but gave up when faced with downloading another burden of ~duplicate packages. (I'm still on 22.11 and couldn't manage to run this flake with `inputs.nixpkgs.follows = "nixpkgs"` at first, and then just had to maintain it with the same unstable revision to avoid further wasting space on hdd. As you can see, the disclaimer is there for a reason!)

Anyway, updating nixpkgs revision instead of overriding those packages may be a solution **for now** (and nixpkgs revision should probably be updated in any case), but this is a pretty endless race! More recent version of InvokeAI again depends on versions missing from stable 23.05. I would really like to not strictly depending on unstable! Is there any good solution to achieve seamless compatibility with both stable and unstable, other than conditionally overriding only if target package is outdated, as I'm trying to do? Maybe long-running release branches, as in nixpkgs itself?